src/boss/lambdaStore.go

```diff
@@ -0,0 +1,300 @@
+		http.Error(w, fmt.Sprintf("Failed to extract config.json: %v", err), http.StatusBadRequest)
+		return
+	}
+
+	s.unregisterTriggers(functionName)
+	s.removeFromRegistry(functionName)
+	s.addToRegistry(functionName, cfg)
+	s.registerTriggers(functionName, cfg)
+
+	w.WriteHeader(http.StatusCreated)
+	fmt.Fprintf(w, "Lambda %s uploaded successfully", functionName)
+}
+
+func (s *LambdaStore) DeleteLambda(w http.ResponseWriter, r *http.Request) {
+	raw := strings.TrimPrefix(r.URL.Path, "/lambda/")
+	functionName, err := sanitizeFunctionName(raw)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	dir := filepath.Join(s.StorePath, functionName)
+	if err := os.RemoveAll(dir); err != nil {
+		http.Error(w, "Failed to delete lambda", http.StatusInternalServerError)
+		return
+	}
+
+	s.unregisterTriggers(functionName)
+	s.removeFromRegistry(functionName)
+
+	w.WriteHeader(http.StatusOK)
+	fmt.Fprintf(w, "Lambda %s deleted successfully", functionName)
+}
+
+func (s *LambdaStore) ListLambda(w http.ResponseWriter, r *http.Request) {
+	files, err := os.ReadDir(s.StorePath)
+	if err != nil {
+		http.Error(w, "Failed to list lambdas", http.StatusInternalServerError)
+		return
+	}
+
+	var names []string
+	for _, file := range files {
+		if file.IsDir() {
+			names = append(names, file.Name())
+		}
+	}
+	json.NewEncoder(w).Encode(names)
+}
+
+func (s *LambdaStore) GetLambdaConfig(w http.ResponseWriter, r *http.Request) {
+	raw := strings.TrimPrefix(r.URL.Path, "/lambda/config/")
+	functionName, err := sanitizeFunctionName(raw)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	tarPath := filepath.Join(s.StorePath, functionName+".tar.gz")
+	f, err := os.Open(tarPath)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to open tarball: %v", err), http.StatusNotFound)
+		return
+	}
+	defer f.Close()
+
+	cfg, err := extractConfigFromTarGz(f)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to extract config: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(cfg); err != nil {
+		http.Error(w, "failed to encode config as JSON", http.StatusInternalServerError)
+		return
+	}
+}
+
+// ------------------- Core Logic ----------------------
+
+func (s *LambdaStore) loadConfigAndRegister(functionName string) error {
+	tarPath := filepath.Join(s.StorePath, functionName+".tar.gz")
+
+	f, err := os.Open(tarPath)
+	if err != nil {
+		return fmt.Errorf("failed to open lambda tarball: %w", err)
+	}
+	defer f.Close()
+
+	cfg, err := extractConfigFromTarGz(f)
+	if err != nil {
+		return fmt.Errorf("failed to extract config.json: %w", err)
+	}
+
+	s.unregisterTriggers(functionName)
+	s.removeFromRegistry(functionName)
+	s.addToRegistry(functionName, cfg)
+	s.registerTriggers(functionName, cfg)
+
+	return nil
+}
+
+func (s *LambdaStore) registerTriggers(functionName string, cfg *common.LambdaConfig) {
```

Comment:
Let's leave the bodies of registerTriggers and unregisterTriggers as TODOs for this PR.  I think events should be a separate subsystem that the registry interacts with instead of having that logic here.  This can eventually end up in boss/event, mirroring worker/event